lib/stripe/plans.rb

```diff
@@ -21,6 +21,36 @@ def initialize(*args)
       end
 
       def create_options
+        if api_version_after_plans_to_product_change
+          default_create_options
+        else
+          pre_product_create_options
+        end
+      end
```

Comment:
I noticed here that in order to accommodate the new API while supporting the previous versions, that some redundancy was introduced. To help DRY up the code, I'd like to suggest the following refactoring. With this refactoring, you can remove the methods `default_create_options` and `pre_product_create_options`. Note I'm also using a suggested method name change from a comment I made in code further down.

```
def create_options
  options =  {
    :currency => @currency,
    :amount => @amount,
    :interval => @interval,
    :interval_count => @interval_count,
    :trial_period_days => @trial_period_days,
    :metadata => @metadata,
    :statement_descriptor => @statement_descriptor
  }
  if api_version_gte_2018_02_05?
    options[:product] = { :name => @name }
  else
    options[:name] = @name
  end
  options
end
```

Reply:
thanks for the suggestion! I appreciate trying to DRY this up but I'm choosing to keep this a little more verbose for clarity.